packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -225,11 +246,22 @@ const fetchAggregationData = (page: number): ThunkAction<
         });
       }
     } catch (e) {
-      dispatch({
-        type: ActionTypes.AggregationFailed,
-        error: (e as Error).message,
-      });
-      log.warn(mongoLogId(1001000106), 'Aggregations', 'Failed to run aggregation');
+      // On cancel, we show the previous state
+      if ((e as Error).name === PROMISE_CANCELLED_ERROR) {
+        dispatch({
+          type: ActionTypes.AggregationFinished,
+          documents: _documents,
+          page: _page,
+          isLast: _documents.length < limit,
+        });
+      } else {
+        dispatch({
+          type: ActionTypes.AggregationFailed,
+          error: (e as Error).message,
+          errorPage: page,
```

Comment:
We already have current page in the state, do we need to duplicate this state for some reason?

Reply:
nice! i ended up removing it and relying on `page`